src/runtime/relax_vm/builtin.cc

```diff
@@ -380,6 +380,40 @@ TVM_REGISTER_GLOBAL("vm.builtin.make_tuple").set_body([](TVMArgs args, TVMRetVal
   *rv = arr;
 });
 
+TVM_REGISTER_GLOBAL("vm.builtin.tensor_to_shape").set_body_typed([](NDArray data) {
+  NDArray arr = data;
+  if (data->device.device_type != kDLCPU) {
+    arr = data.CopyTo(DLDevice{kDLCPU, 0});
```

Comment:
Does this copy fail if the NDArray is on another device? I'm a little hesitant to have an op that just will not work depending on the device.

Reply:
Is it possible that we have a shape tensor not on host device?

Reply:
Not sure where this might come up. I wouldn't be surprised if, say, we import an ONNX model (this is where this use-case first came up) that we mean to run on GPU and every tensor (including those that stand for shapes) is stored on GPU.

Reply:
Based on my current understanding, we want to keep the shape-related computation on the host side since its result could be related to the memory planning. 
Even for general GPU execution besides this case, we are running output shape computation (inserted by `VMShapeLower` pass) on the host side.